### DISCUSSION TITLE
Using both wagmi react and wagmi core in one app

### DISCUSSION THREAD

**Discussion by nezouse** [wevm/wagmi#1766] [Ideas]
Hello, i would like to use both react wrapper and core client in same app. I'm fetching data from multiple sources and i wanted to wrap it in tanstack query by myself, but seems like i have to initialise client twice. Could something be done that after initialising wagmi react core is also initialized? Or maybe i was doing something wrong and it should be working like that by default?

**Comment by tmm** [comment]
You shouldn't need to set up the client twice. Just set it up once and import `@wagmi/core` actions from `wagmi/actions` entrypoint instead:

```ts
import { fetchBlockNumber } from 'wagmi/actions'

await fetchBlockNumber()
```

**Reply by nezouse** [reply]
Is it provided anywhere in docs? I was using actions from wagmi/core and in this case it wasn't working properly.

**Reply by tmm** [reply]
Added a page to the docs site: https://wagmi.sh/react/actions
